test/src/unit-cppapi-profile.cc

```diff
@@ -0,0 +1,86 @@
+/**
+ * @file   unit-cppapi-profile.cc
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2025 TileDB Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * Tests the C++ API for profile-related functionality.
+ */
+
+#include <test/support/src/helpers.h>
+#include <test/support/tdb_catch.h>
+
+#include "test/support/src/temporary_local_directory.h"
+#include "tiledb/sm/cpp_api/profile_experimental.h"
+#include "tiledb/sm/rest/rest_profile.h"
+
+using namespace tiledb;
+using namespace tiledb::test;
+
+struct ProfileCPPFx {
+  const std::string name_;
+  tiledb::sm::TemporaryLocalDirectory tempdir_;
+
+  ProfileCPPFx()
+      : name_(tiledb::sm::RestProfile::DEFAULT_NAME)
+      , tempdir_("unit_cppapi_profile") {
+  }
+};
+
+TEST_CASE_METHOD(
+    ProfileCPPFx,
+    "C++ API: Profile get_name validation",
+    "[cppapi][profile][get_name]") {
+  const std::string homedir_ = tempdir_.path();
+  SECTION("default, explicitly passed") {
+    Profile p(name_, homedir_);
+    REQUIRE(p.get_name() == name_);
+  }
+  SECTION("default, inherited from nullptr") {
+    Profile p(std::nullopt, homedir_);
+    REQUIRE(p.get_name() == name_);
+  }
+  SECTION("non-default") {
+    const char* name = "non_default";
+    Profile p(name, homedir_);
+    REQUIRE(p.get_name() == name);
+  }
+}
+
+TEST_CASE_METHOD(
+    ProfileCPPFx,
+    "C++ API: Profile get_homedir validation",
+    "[cppapi][profile][get_homedir]") {
+  auto homedir_ = tempdir_.path().c_str();
+  SECTION("explicitly passed") {
+    Profile p(name_, homedir_);
+    REQUIRE(p.get_homedir() == homedir_);
+  }
+  SECTION("inherited from nullptr") {
+    Profile p(name_, std::nullopt);
+    REQUIRE(p.get_homedir() == tiledb::common::filesystem::home_directory());
```

Comment:
Is the "homedir" `$HOME/.tiledb` or just `$HOME`?
And whatever path you put in gets `/.tiledb` appended?
I think I have my answer here:
```
RestProfile::RestProfile(const std::string& name, const std::string& homedir)
    : version_(constants::rest_profile_version)
    , name_(name)
    , filepath_(homedir + constants::rest_profile_filepath)
    , old_filepath_(homedir + constants::cloud_profile_filepath) {
  // Fstream cannot create directories. If `homedir/.tiledb/` DNE, create it.
  std::filesystem::create_directories(homedir + ".tiledb");
  ...
}
```

This strikes me as a little too specific.  Rather than forcing the notion of a home directory " I feel like this ought to be a "profile directory" whose default is `$HOME/.tiledb` and we don't append anything.

Reply:
I'm marking this comment as one which I think we should discuss but not one which should hold up this PR.  Address it now if you agree; if you don't then let's focus on the other comments and circle back here once this PR is merged.

Reply:
- `homedir` is `$HOME`.
- `constants::rest_profile_filepath` is `".tiledb/profiles.json"`.

If i understand correctly, you are suggesting to remove the `.tiledb` from the constant and move it to `homedir`?

Reply:
Not quite. What I'm suggesting is more like
```suggestion
    REQUIRE(p.get_homedir() == tiledb::common::filesystem::home_directory().join(".tiledb"));
```